### ISSUE TITLE
CustomTestSimple.js not working with RedLine

### ISSUE THREAD

**Issue by Joralf** [redline13/harness-custom-test-nodejs#1]
Hi there!

Thanks so much for sharing a few great methods to do testing via RedLine. I also love the ability to run tests on my local machine. I do however have a question: I figured, since all code was working locally I could try doing a test in RedLine as well. 

If I see correctly `loadTester.js` and `redlineApi.js` are there for local purposes only. So I uploaded a tar file containing `CustomTest.js` (CustomTestSimple.js in repo), `loadtest.ini`, `loadTestingHttpRequest.js`, `loadTestingSession.js` and `package.json`.

But whatever I do (or try to change) I keep getting the following errors:

`Failed to launch: TypeError: object is not a function TypeError: object is not a function at Object.LoadTester.runTest (/home/user32986/loadTester.js:89:15) at null._onTimeout (/home/user32986/loadTester.js:75:10) at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)`

`AssertionError: path must be a string AssertionError: path must be a string at Module.require (module.js:362:3) at require (module.js:380:17) at Object.LoadTester.runTests (/home/user32986/loadTester.js:65:16) at Object. (/home/user32986/loadTester.js:126:12) at Module._compile (module.js:456:26) at Object.Module._extensions..js (module.js:474:10) at Module.load (module.js:356:32) at Function.Module._load (module.js:312:12) at Function.Module.runMain (module.js:497:10) at startup (node.js:119:16)`

If you're able to reproduce this, my first question would be: did something change in the RedLine API/tester and was this change not applied to this repo, hence why it breaks? I think we can go on from there if that is the case. If not, do you have any pointers I could look at to see why this breaks?

Thanks so much! Looking forward to the response and wouldn't mind contributing if some code needs to be changed :)

**Comment by richardfriedman** [comment]
@Joralf thanks for this issue. 

You only need to include 
- your custom test source (CustomTestSimple.js in this case)
- package.json - if you need any extra packages

In your case the issue is probably loadtest.ini, redline13.com generates this with settings which are required for the production version.  Other files like loadTesting*.js should also not be included. 

I added a section in the ReadMe - Running on RedLine13.com.

**Comment by Joralf** [comment]
Hi Richard! 

Thanks for the quick reaction. So if I summarize correctly: all files except for CustomTestSimple.js are mocked versions of methods that are part of the RedLine testsuite?

So if I would want to use loadTestingSession, I can just include like you did on line 1 of CustomTestSimple.js and RedLine will include these modules during runtime? Or do I need to extract the methods of my liking and put them in the CustomTestSimple prototype?

All the best,

Joralf

**Comment by richardfriedman** [comment]
> So if I summarize correctly: all files except for CustomTestSimple.js are mocked versions of methods that are part of the RedLine testsuite?

Correct. Though you can include own package.json

> and RedLine will include these modules during runtime? 

Correct, we already include loadTestingHttpRequest.js and loadTestingSession.js.  
If you include your own (technically you can) it will override the current ones we are using.   We should really move these to their own npm module (TODO)

**Comment by Joralf** [comment]
Great! It's really helpful if we can use those methods during tests. Thanks so much for the response.

To be a bit more precise about my confusion. I think this is were I went wrong:

> You can include the helper class loadTestingSession for testing HTTP requests.  This class helps manage cookies and track performance data.

`var LoadTestingSession = require("./loadTestingSession.js");`

from: https://www.redline13.com/blog/custom-load-test-nodejs/

In which I assumed I had to add loadTesting in the tar-file because I somehow didn't get the memo that all files are mockups of the RedLine13 test suite.
